metrics/cli/lib/sentry/adapter/sentry_cli_service_adapter.dart

```diff
@@ -0,0 +1,120 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:cli/helper/file_helper.dart';
+import 'package:cli/prompt/prompter.dart';
+import 'package:cli/sentry/cli/sentry_cli.dart';
+import 'package:cli/sentry/constants/sentry_constants.dart';
+import 'package:cli/sentry/model/sentry_project.dart';
+import 'package:cli/sentry/service/sentry_service.dart';
+import 'package:cli/sentry/strings/sentry_strings.dart';
+
+/// An adapter for the [SentryCli] to implement the [SentryService] interface.
+class SentryCliServiceAdapter implements SentryService {
+  /// A [SentryCli] class that provides an ability to interact
+  /// with the Sentry CLI.
+  final SentryCli _sentryCli;
+
+  /// A [Prompter] class this adapter uses to interact with a user.
+  final Prompter _prompter;
+
+  /// A [FileHelper] class this adapter uses to work with file system.
+  final FileHelper _fileHelper;
+
+  /// Creates a new instance of the [SentryCliServiceAdapter]
+  /// with the given [sentryCli], [prompter], and [fileHelper].
+  ///
+  /// Throws an [ArgumentError] if the given [sentryCli] is `null`.
+  /// Throws an [ArgumentError] if the given [prompter] is `null`.
+  /// Throws an [ArgumentError] if the given [fileHelper] is `null`.
+  SentryCliServiceAdapter({
+    SentryCli sentryCli,
+    Prompter prompter,
+    FileHelper fileHelper,
+  })  : _sentryCli = sentryCli,
+        _prompter = prompter,
+        _fileHelper = fileHelper {
+    ArgumentError.checkNotNull(_sentryCli, 'sentryCli');
+    ArgumentError.checkNotNull(_prompter, 'prompter');
+    ArgumentError.checkNotNull(_fileHelper, 'fileHelper');
+  }
+
+  @override
+  Future<void> login() {
+    return _sentryCli.login();
+  }
+
+  @override
+  Future<void> createRelease(
+    String appPath,
+    String buildPath,
+    String configPath,
+  ) async {
+    final sentryProject = _getSentryProject();
+    final releaseName = _prompter.prompt(SentryStrings.enterReleaseName);
+
+    await _sentryCli.createRelease(releaseName, sentryProject);
+    await _uploadSourceMaps(appPath, buildPath, sentryProject, releaseName);
+    await _sentryCli.finalizeRelease(releaseName, sentryProject);
+    _updateConfig(sentryProject, configPath, releaseName);
```

Comment:
It seems like the `SentryService` should not know anything about the Metrics Web application and its configuration. I think this method should only create a new Sentry release. What do you think?